codalab/server/bundle_manager.py

```diff
@@ -327,6 +330,11 @@ def _deduct_worker_resources(self, workers_list):
         """
         for worker in workers_list:
             for uuid in worker['run_uuids']:
+                # If the resource of uuid has been subtracted before, continue to next uuid.
```

Comment:
We should also check whether this bundle still exists here. Currently there's a potential race condition where the bundle might get deleted while the bundle manager execution is right between the lines 314 and 316 of this file (ie after we fetch the workers list from the DB and before we call deduct worker resources)

This race condition is causing #1572 

I guess ideally we'd use something like a session (as suggested by #1504) to make sure the bundles don't get deleted while we're in the middle of reading that table, but the alternate solution of checking for bundle existence here seems good enough for now.

Reply:
@bkgoksel Do you mean to change the return of this function? https://github.com/codalab/codalab-worksheets/blob/master/codalab/model/bundle_model.py#L171